src/Parser.js

```diff
@@ -38,28 +38,50 @@ import ParseError from "./ParseError";
  *
  * The earlier functions return ParseNodes.
  * The later functions (which are called deeper in the parse) sometimes return
- * ParseFuncOrArgument, which contain a ParseNode as well as some data about
+ * ParsedFuncOrArgOrDollar, which contain a ParseNode as well as some data about
  * whether the parsed object is a function which is missing some arguments, or a
  * standalone object which can be used as an argument to another function.
  */
 
-/** A function name or an argument to a function. */
-class ParseFuncOrArgument {
-    /**
-     * @param {ParseNode|string} result It's a string if `isFunction=true` or if
-     *     `result="$"` for switching into math mode; otherwise, it's a ParseNode
-     *     and `isFunction` must be false. If it's a function, the string should
-     *     be a name defined by defineFunction, e.g. "\\frac".
-     * @param {boolean} isFunction True when this is a function. False when it's a
-     *     function argument or "$" to switch into math mode.
-     * @param {Token} token
-     */
-    constructor(result, isFunction, token) {
-        this.result = result;
-        // Is this a function (i.e. is it something defined in functions.js)?
-        this.isFunction = isFunction;
-        this.token = token;
+/* TODO: Uncomment when porting to flow.
+type ParsedType = "fn" | "arg" | "$"
+type ParsedFunc = {|
+    type: "fn",
+    result: string, // Function name defined via defineFunction (e.g. "\\frac").
+    token?: ?Token,
+|};
+type ParsedArg = {|
+    type: "arg",
+    result: ParseNode,
+    token?: ?Token,
+|};
+type ParsedDollar = {|
+    // Math mode switch
+    type: "$",
+    result: "$",
+    token?: ?Token,
+|};
+type ParsedFuncOrArgOrDollar = ParsedFunc | ParsedArg | ParsedDollar;
+*/
+
+/**
+ * @param {ParseNode} result
+ * @param {?Token=} token
+ * @return {ParsedArg}
+ */
+function newArgument(result, token) {
+    return {type: "arg", result, token};
+}
```

Comment:
I would be nice to have symmetry between how `ParsedFunc`, `ParsedArg`, and `ParsedDollar` get created.

Reply:
Done. I was just keeping the generated code low a concern you'd mentioned in a previous PR.

Reply:
I thinking I was being overly picky in that previous PR, sorry about that.  We definitely want to keep the size down, but I doubt this will have much of an impact especially after uglify.  As we continue to make things more explicit in Parser.js I'm sure the code size will fluctuate up and down a bit.  In the other diff you dedupe `parseArguments` so we saved some bytes so it's all good.  ^_^